anchore_engine/services/policy_engine/engine/feeds/download.py

```diff
@@ -127,10 +123,63 @@ def read(self, feed, group, start_index):
         ]
         return FileListJsonIterator(files, start_index)
 
+    def _get_group_metadata(self, feed, group):
+        for results in self.metadata.download_result.results:
+            if results.feed == feed and results.group == group:
+                return results.group_metadata
+
+    def read_files(self, feed, group):
+        files = [
+            os.path.join(self.root_dir, feed, group, x)
+            for x in os.listdir(os.path.join(self.root_dir, feed, group))
+        ]
+        group_metadata = self._get_group_metadata(feed, group)
+        return FileListIterator(files, group_metadata)
+
     def teardown(self):
         shutil.rmtree(self.root_dir)
 
 
+@dataclass
+class FileData:
+    data: bytes
+    metadata: Dict[str, str]
+
+
+class FileListIterator(object):
+    """
+    Iterator that will return a FileData object from files listed in order handling the open/close between files.
+    """
+
+    def __init__(
+        self, ordered_path_list: list, group_metadata: Dict[str, Dict[str, str]]
+    ):
+        self.files = [
+            file_path for file_path in ordered_path_list if os.path.isfile(file_path)
+        ]
+        self.file_metadata = group_metadata
+        self.current_file_index = 0
+
+    def __iter__(self):
+        return self
+
+    def _read_file(self, file) -> FileData:
+        name = os.path.split(file)[1]
+        with open(file, "rb") as f:
+            return FileData(f.read(), self.file_metadata[name])
+
+    def _get_currentfile(self):
+        return self.files[self.current_file_index]
+
+    def __next__(self) -> FileData:
+        if self.current_file_index < len(self.files):
+            file_data = self._read_file(self._get_currentfile())
+            self.current_file_index += 1
+            return file_data
+        else:
```

Comment:
There is no need for this `else:` condition here, the `StopIteration()` will follow the same logic, and can be outdented

Reply:
whoops. i'll fix that. thanks for catching